### PR TITLE
Image Rendering: Add settings for default width, height and scale

### PR DESCRIPTION
**Pull request by khushijain21**
This feature provides the ability to configure image width, height and scale for rendering images 

Part of https://github.com/grafana/grafana/issues/75963

### CODE REVIEW

File: conf/defaults.ini
Comment (AgnesToulet) [inline conf/defaults.ini:1559]:
```suggestion
# Default width for panel screenshot
default_image_width = 1000
# Default height for panel screenshot
default_image_height = 500
# Default scale for panel screenshot
default_image_scale = 1
```

File: pkg/setting/setting.go
Comment (AgnesToulet) [inline pkg/setting/setting.go:1775]:
```suggestion
	cfg.RendererDefaultImageWidth = renderSec.Key("default_image_width").MustInt(1000)
	cfg.RendererDefaultImageHeight = renderSec.Key("default_image_height").MustInt(500)
	cfg.RendererDefaultImageScale = renderSec.Key("default_image_scale").MustInt(1)
```

File: pkg/setting/setting.go
Comment (AgnesToulet) [inline pkg/setting/setting.go:153]:
```suggestion
	RendererDefaultImageWidth             int
	RendererDefaultImageHeight            int
	RendererDefaultImageScale             int
```

File: pkg/api/render.go
Comment (AgnesToulet) [inline pkg/api/render.go:31]: I think this should be refactored to something like:
```
	width := c.QueryInt("width")
	if width == 0 {
		width = hs.Cfg.RendererDefaultImageWidth
	}
```
Because it makes more sense to have the default values as int rather than string.

**Review by AgnesToulet — COMMENTED** [review]
Thank you for your contribution on this issue. I left a couple of suggestions, mostly around naming to make it clear these are defaults and can be overridden. 

This would be nice to also add these new setting fields in [the documentation](https://github.com/grafana/grafana/blob/main/docs/sources/setup-grafana/configure-grafana/_index.md#rendering). 

Also, this doesn't really fix the initial issue because the values are still hard-coded in the frontend and override these defaults. They should be removed from [the frontend](https://github.com/grafana/grafana/blob/f1c3177e79f21e1292afa02efa393f5e52dc231a/public/app/features/dashboard/components/ShareModal/utils.ts#L106).

File: conf/defaults.ini
Comment (AgnesToulet) [inline conf/defaults.ini:1541]:
```suggestion
server_url =
```

File: conf/defaults.ini
Comment (AgnesToulet) [inline conf/defaults.ini:1543]:
```suggestion
callback_url =
```

File: conf/defaults.ini
Comment (AgnesToulet) [inline conf/defaults.ini:1559]:
```suggestion
default_image_scale = 1

```

File: docs/sources/setup-grafana/configure-grafana/_index.md
Comment (AgnesToulet) [inline docs/sources/setup-grafana/configure-grafana/_index.md:2182]:
```suggestion
### default_image_width

To configure the width of the rendered image. The default width is 1000.

### default_image_height

To configure the height of the rendered image. The default height is 500.

### default_image_scale

To configure the scale of the rendered image. The default scale is 1.

```

File: pkg/api/dtos/frontend_settings.go
Comment (AgnesToulet) [inline pkg/api/dtos/frontend_settings.go:213]:
```suggestion
	RendererDefaultImageWidth               int                            `json:"rendererDefaultImageWidth"`
	RendererDefaultImageHeight              int                            `json:"rendererDefaultImageHeight"`
	RendererDefaultImageScale               float64                        `json:"rendererDefaultImageScale"`
```

File: packages/grafana-runtime/src/config.ts
Comment (AgnesToulet) [inline packages/grafana-runtime/src/config.ts:104]:
```suggestion
  rendererDefaultImageWidth = 1000;
  rendererDefaultImageHeight = 500;
  rendererDefaultImageScale = 1;
```

File: pkg/api/frontendsettings.go
Comment (AgnesToulet) [inline pkg/api/frontendsettings.go:249]:
```suggestion
		RendererDefaultImageWidth:               hs.Cfg.RendererDefaultImageWidth,
		RendererDefaultImageHeight:              hs.Cfg.RendererDefaultImageHeight,
		RendererDefaultImageScale:               hs.Cfg.RendererDefaultImageScale,
```

File: pkg/api/render.go
Comment (AgnesToulet) [inline pkg/api/render.go:32]:
```suggestion
	height := c.QueryInt("height")
```

File: pkg/api/render.go
Comment (AgnesToulet) [inline pkg/api/render.go:43]: You can add this in `pkg/web/context.go`:
```
// QueryFloat64 returns query result in float64 type.
func (ctx *Context) QueryFloat64(name string) float64 {
	n, _ := strconv.ParseFloat(ctx.Query(name), 64)
	return n
}
```

File: pkg/api/render.go
Comment (AgnesToulet) [inline pkg/api/render.go:75]:
```suggestion
		DeviceScaleFactor: scale,
```

File: public/app/features/dashboard/components/ShareModal/utils.ts
Comment (AgnesToulet) [inline public/app/features/dashboard/components/ShareModal/utils.ts:127]:
```suggestion
    `&width=${config.rendererDefaultImageWidth}` +
    `&height=${config.rendererDefaultImageHeight}` +
    `&scale=${config.rendererDefaultImageScale}` +
```

File: pkg/api/frontendsettings.go
Comment (AgnesToulet) [inline pkg/api/frontendsettings.go:249]: Spacing should be fixed to pass linter check.

File: pkg/api/dtos/frontend_settings.go
Comment (AgnesToulet) [inline pkg/api/dtos/frontend_settings.go:213]: Spacing should be fixed to pass linter check.

**Review by AgnesToulet — APPROVED** [review]
LGTM! Thank you very much for your contribution.

Some spacing should be fixed to pass the linter check in the CI pipeline, and I'll wait for the doc squad feedback before merging this, but code-wise, everything looks good to me.

File: docs/sources/setup-grafana/configure-grafana/_index.md
Comment (AgnesToulet) [inline docs/sources/setup-grafana/configure-grafana/_index.md:2171]: @chri2547 Could you take a look before we merge, please?

Comment (chri2547) [reply docs/sources/setup-grafana/configure-grafana/_index.md:2171]: @AgnesToulet , on it! Thank you.

File: docs/sources/setup-grafana/configure-grafana/_index.md
Comment (chri2547) [inline docs/sources/setup-grafana/configure-grafana/_index.md:2173]:
```suggestion
Configures the width of the rendered image. The default width is `1000`.
```

File: docs/sources/setup-grafana/configure-grafana/_index.md
Comment (chri2547) [inline docs/sources/setup-grafana/configure-grafana/_index.md:2177]:
```suggestion
Configures the height of the rendered image. The default height is `500`.
```

File: docs/sources/setup-grafana/configure-grafana/_index.md
Comment (chri2547) [inline docs/sources/setup-grafana/configure-grafana/_index.md:2181]:
```suggestion
Configures the scale of the rendered image. The default scale is `1`.
```

**Review by chri2547 — COMMENTED** [review]
A few slight adjustments to docs. Thank you for the contribution! If the changes look ok to you, please accept them and then ping me for approval.

**Review by chri2547 — APPROVED** [review]
Docs approved, thank you!

**Comment by AgnesToulet** [comment]
Hi! Sorry for the delay, but could you update your branch with the latest main? It should make the CI pipeline succeed, and I'll merge the PR then.

**Pull request merged**
